Sources/Plasma/FeatureLib/pfPython/pyImageGlue.cpp

```diff
@@ -165,21 +165,35 @@ PYTHON_METHOD_DEFINITION(ptImage, saveAsJPEG, args)
 
 PYTHON_METHOD_DEFINITION(ptImage, saveAsPNG, args)
 {
-    PyObject* filenameObj;
-    if (!PyArg_ParseTuple(args, "O", &filenameObj))
+    PyObject* filenameObj, *optionalText;
+    if (!PyArg_ParseTuple(args, "O|O", &filenameObj, &optionalText))
     {
-        PyErr_SetString(PyExc_TypeError, "saveAsPNG expects a string");
+        PyErr_SetString(PyExc_TypeError, "saveAsPNG expects a string, and an optional dict of key-value pairs");
         PYTHON_RETURN_ERROR;
     }
 
     if (PyString_CheckEx(filenameObj))
     {
-        self->fThis->SaveAsPNG(PyString_AsStringEx(filenameObj));
+        if (PyDict_Check(optionalText)) {
+            auto textFields = new std::multimap<plString, plString>();
+            PyObject *key, *value;
+            Py_ssize_t pos = 0;
+            while (PyDict_Next(optionalText, &pos, &key, &value)) {
+                if (PyString_Check(key) && PyString_Check(value)) {
```

Comment:
For reference, `ST::string`'s `to_latin1()` converter allows you to replace invalid characters during conversion (instead of asserting):
https://github.com/zrax/string_theory/blob/master/doc/st_string.md#to_latin_1